### PR TITLE
Bug 1834621: add validation to traffic modal

### PR DESCRIPTION
**Pull request by nemesis09**
Fixes:
https://issues.redhat.com/browse/ODC-3355

Analysis / Root cause:
Including a period "." in the tag causes the traffic distribution action to not work. Although there is no error messages or notification to the user to let them know

Solution Description:
add proper validation to traffic modal

Screens:
Screenshot from 2020-05-12 09-47-34
Screenshot from 2020-05-12 09-48-07

Code Coverage:
Screenshot from 2020-05-12 09-49-17

Browser Conformance:
- [x] Firefox
- [x] Chrome
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by nemesis09** [comment]
The error message overlaps on the fields below. should we use a shorter error message?
@serenamarie125 @beaumorley @openshift/team-devconsole-ux 
Screenshot from 2020-05-12 09-47-34

File: frontend/packages/knative-plugin/src/utils/trafficSplitting-validation-utils.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        .required('Required')
+        .max(100, 'cannot exceed 100%'),
+      tag: yup
+        .string()
+        .matches(nameRegex, {
+          message:
+            'tag name must consist of lower-case letters, numbers and hyphens. It must start with a letter and end with a letter or number.',
+          excludeEmptyString: true,
+        })
+        .max(253, 'Cannot be longer than 253 characters.'),
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/utils/trafficSplitting-validation-utils.ts:19]: Do we really want 253 character's here

Comment (nemesis09) [reply frontend/packages/knative-plugin/src/utils/trafficSplitting-validation-utils.ts:19]: Not really, I wasn't sure how many characters should be allowed, so just kept the standard limit maintained for names across the app

**Comment by nemesis09** [comment]
cloasing off the PR as it has been decided to go ahead with server side validation alone. Please refer to the conversation for more details.
https://coreos.slack.com/archives/GG5NMJEJH/p1589300990459600

**Pull request closed without merging**